src/Features/CSharp/Portable/ConvertLinq/CSharpConvertLinqQueryToForEachProvider.cs

```diff
@@ -0,0 +1,792 @@
+                                  SyntaxFactory.ArgumentList(),
+                                  initializer: null),
+                               variableName: "list",
+                              out documentUpdateInfo);
+                }
+
+                documentUpdateInfo = default;
+                return false;
+            }
+
+            private bool TryConvertIfInInvocation(
+                QueryExpressionSyntax source,
+                InvocationExpressionSyntax invocationExpression,
+                Func<ExpressionSyntax, ExpressionSyntax, StatementSyntax> leafExpressionCreationMethod,
+                ExpressionSyntax initializer,
+                string variableName,
+                out DocumentUpdateInfo documentUpdateInfo)
+            {
+                var parentStatement = invocationExpression.GetAncestorOrThis<StatementSyntax>();
+                if (parentStatement != null)
+                {
+                    if (TryConvertIfInInvocationInternal(
+                            source,
+                            invocationExpression,
+                            parentStatement,
+                            initializer,
+                            variableName,
+                            out var variable,
+                            out var documentCreationMethod))
+                    {
+                        if (TryConvertQueryExpression(
+                            source,
+                            expression => leafExpressionCreationMethod(variable, expression),
+                            out var statement))
+                        {
+                            documentUpdateInfo = documentCreationMethod(statement);
+                            return true;
+                        }
+                    }
+                }
+
+                documentUpdateInfo = default;
+                return false;
+            }
+
+            private bool TryConvertIfInInvocationInternal(
+                QueryExpressionSyntax source,
+                InvocationExpressionSyntax invocationExpression,
+                StatementSyntax parentStatement,
+                ExpressionSyntax initializer,
+                string variableName,
+                out ExpressionSyntax variable,
+                out Func<StatementSyntax, DocumentUpdateInfo> documentUpdateMethod)
+            {
+                var invocationParent = invocationExpression.Parent;
+                switch (invocationParent.Kind())
+                {
+                    case SyntaxKind.EqualsValueClause:
+                        if (invocationParent.Parent is VariableDeclaratorSyntax variableDeclarator)
+                        {
+                            // before var a = (from a in b select a).ToList();
+                            // after var a = new List<T>();
+                            // foreach(...)
+                            variable = SyntaxFactory.IdentifierName(variableDeclarator.Identifier);
```

Comment:
>variable = SyntaxFactory.IdentifierName(variableDeclarator.Identifier); [](start = 28, length = 71)

Why do we assume that the variable is ```int``` for Count and ```List<T>``` for ToList? It could be some user-defined type with an implicit conversion from int/List and it might have no ```++``` operator or Add method.